runtime/src/main/java/org/corfudb/runtime/collections/TxnStreamingManager.java

```diff
@@ -148,7 +154,13 @@ private synchronized void scheduleTxnPolling() {
         if (!lockedStreamingSubscriptionContexts.isEmpty()) {
             log.trace("Locked {} StreamingSubscriptionContexts for processing",
                     lockedStreamingSubscriptionContexts.size());
-            pollerExecutor.submit(new TransactionPoller(runtime, lockedStreamingSubscriptionContexts));
+
+            // To prevent leaking transaction stream caches, explicitly remove the prior entry
+            if (transactionPoller != null) {
```

Comment:
I think we need to get rid of this. Why create a new stream on every poll ?

If you really need to create a new stream object on every poll, then get an unsafe stream reference, so that when you create one, the JVM GC will GC it.

Reply:
fixed, please help review latest patch. thanks!